### PR TITLE
sql: add new timestamp format for Sequelize

### LINKED ISSUE #13157: sql: can't parse timestamp stored by Sequelize
repro (queries generated by Sequelize):

```
CREATE TABLE IF NOT EXISTS "user" ("id"  SERIAL , "id2" INTEGER , "first_name" VARCHAR(255), "lastName" VARCHAR(255), "phone" VARCHAR(255), "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL, "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id","id2"));

INSERT INTO "user" ("id","id2","first_name","lastName","phone","createdAt","updatedAt") 
VALUES (1,1,'John','Hancock','911','2017-01-26 01:49:31.319 +00:00','2017-01-26 01:49:31.319 +00:00');
```

Note that the values of `updatedAt` and `createdAt` are in the format `2017-01-26 01:49:31.319 +00:00`.

This is a straightforward date + time (with fractional seconds) + timezone offset.

### PR DESCRIPTION
**Pull request by cuongdo**
Resolves #13157

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13159)

### CODE REVIEW

File: pkg/sql/parser/datum.go [hunk 1]
```diff
@@ -963,6 +963,7 @@ const (
 	timestampWithOffsetSecondsZoneFormat  = timestampWithOffsetZoneFormat + ":00"
 	timestampWithNamedZoneFormat          = timestampFormat + " MST"
 	timestampRFC3339NanoWithoutZoneFormat = "2006-01-02T15:04:05"
+	timestampSequelizeFormat              = timestampFormat + ".000 -07:00"
```
Comment (cuongdo) [inline pkg/sql/parser/datum.go]: I’m not a huge fan of this name, but I couldn’t stomach following the convention with `timestampWithFractionalSecondsAndOffsetSecondsZoneFormat`

Comment (tamird) [inline pkg/sql/parser/datum.go]: Meh, this constant is referenced in one place; I'd go for the long name.

**Review by tamird — APPROVED** [review]

**Review by jordanlewis — APPROVED** [review]
LGTM! I'm impartial about that constant name.

**Review by dt — APPROVED** [review]
LGTM.

I have no problem with either name -- if we changed it, I'd probably add a comment indicating which ORM it came from instead, but I don't feel strongly about it either way.

**Pull request merged**